### PR TITLE
PWA-2419: Support new branding terms

### PR DESCRIPTION
**Pull request by michaelyu0123**
- Updated file extensions from **.ee.** to **.ac.** and **.ce.** to **.mos.**
- Updated EE to AC and CE to MOS for file documentations and code while maintaining backward compatiblility

## Description

Overview

The Community Edition (CE) and Enterprise Edition (EE) of our software have been rebranded as "Magento Open Source" and "Adobe Commerce" respectively.

We should support these new branding terms in PWA Studio for consistency.

Please note that renaming repositories is out of scope.

## Related Issue

Closes #PWA-2419

## Acceptance

Acceptance Criteria

Accepted values for MAGENTO_BACKEND_EDITION environment variable are updated
"AC" is added and behaves like "EE"
"MOS" is added and behaves like "CE"
Documentation is updated for this change
All files with ".ee.js" extension are updated to ".ac.js"
All code that references ".ee.js" is updated to behave the same way when ".ac.js" is encountered
packages/pwa-buildpack/lib/WebpackTools/MagentoResolver.js
graphql.config.js
packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js
packages/pwa-buildpack/lib/WebpackTools/configureWebpack/configureWebpack.js
packages/pwa-buildpack/envVarDefinitions.json
Project scaffolding prompts: packages/create-pwa/lib/index.js
others?
All files with ".ce.js" extension are updated to ".mos.js"
All code that references ".ce.js" is updated to behave the same way when ".mos.js" is encountered
Documentation is updated

#### Test scenario(s) for direct fix/feature

Magento Backend Edition environment variable
If set to "AC", app behaves like current "EE", and EE-specific features like Wishlist actions dialog are enabled
If set to "MOS", app behaves like current "CE" and EE-specific features like Wishlist actions dialog are disabled
Verify above for Scaffolded project as well.
Run cypress for PR deployed instance.
Verify default nature of variable (I think its community edition by default) has not changed
Adobe Commerce specific files
Update a file with extension ".ee.js" to ".ac.js" and ensure you are served that version when MAGENTO_BACKEND_EDITION is set to "EE" or "AC".

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run pwa-cypress-test

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Comment by dpatil-magento** [comment]
run cypress

**Comment by supernova-at** [comment]
I'm hoping this can be a `Minor` change. Is there anything specific you're thinking that is backwards incompatible?

**Comment by michaelyu0123** [comment]
> I'm hoping this can be a `Minor` change. Is there anything specific you're thinking that is backwards incompatible?

@supernova-at, oops yes this probably should be a Minor level change since the main issue is if users imported any of our files that uses *.ee.js or *.ce.js the import would be broken since the file names have been changed and that would just be BIC and not an APi change. Updating tag accordingly.

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by yaroslav-zenin** [comment]
This renaming will makes PWA upgrade more dificult for me
we have lot of projects with overrides, targetables, some also patches(

**Comment by michaelyu0123** [comment]
> This renaming will makes PWA upgrade more dificult for me we have lot of projects with overrides, targetables, some also patches(

@supernova-at @yaroslav-zenin  if that is the case would it make more sense to skip the file renaming and just add support for ac and mos extensions?

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
@supernova-at all the file renames have been reverted. Graphql, create-pwa, webpack, and Magento Resolver should all still support CE and EE version. However will give priority to AC and MOS extensions if both extensions are found.

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

File: packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ export const useGiftCards = props => {
  * @property {GraphQLAST} getAppliedGiftCardsQuery The query used to get the gift cards currently applied to the cart.
  * @property {GraphQLAST} getGiftCardBalanceQuery The query used to get the gift cards currently applied to the cart.
  *
- * @see [`giftCardQueries.ee.js`]{@link https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/GiftCards/giftCardQueries.js}
+ * @see [`giftCardQueries.ee.js`]{@link https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/CartPage/GiftCards/giftCardQueries.gql.ee.js}
```
Comment (jeremycharron) [inline packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js:204]: Is it related to the story?
I'm not sure to understand the change.

File: packages/venia-ui/lib/targets/__tests__/venia-ui-targets.spec.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const thisDep = {
     intercept
 };
 
-const WEBPACK_BUILD_TIMEOUT = 20000;
+const WEBPACK_BUILD_TIMEOUT = 30000;
```
Comment (jeremycharron) [inline packages/venia-ui/lib/targets/__tests__/venia-ui-targets.spec.js:17]: It this related to story ?

Comment (michaelyu0123) [reply packages/venia-ui/lib/targets/__tests__/venia-ui-targets.spec.js:17]: it is not, but it helps stabilize this test when run locally.

**Review by jeremycharron — APPROVED** [review]
Approved. Ready for QA.

**Comment by ericeoeur** [comment]
@michaelyu0123 , there are some conflicts that have emerged above. Can you please review/resolve?

**Comment by ericeoeur** [comment]
When testing the following PR with a CE/MOS [Magento backend](https://pwa-test-djwdmpa-vqnoqmbf7mivy.us-4.magentosite.cloud/), a bug was found as homepage styling is broken. This bug has been reported in the backlog as [PWA-2629](https://jira.corp.magento.com/browse/PWA-2629).

**Comment by michaelyu0123** [comment]
yarn run tests

**Comment by michaelyu0123** [comment]
run all tests

**Review by mikhaelbois — DISMISSED** [review]
LGTM ✅

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Review by ericeoeur — APPROVED** [review]
LGTM

**Comment by ericeoeur** [comment]
QA approved, will merge momentarily.

**Pull request merged**